### PR TITLE
feature(InterPodAffinity): return Skip in PreFilter

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

return Skip in InterPodAffinity PreFilter when **both** of the following condition is satisfied:
- no PodAffinity and PodAntiAffinity on an incoming Pod.
- no PodAntiAffinity on existing Pod which match with an incoming Pod.

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/kubernetes/issues/114399

#### Does this PR introduce a user-facing change?

```release-note
Scheduler skips InterPodAffinity Filter plugin when InterPodAffinity Filter plugin has nothing to do with a Pod.
It may affect some metrics values related to the InterPodAffinity Filter plugin.
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go [hunk 1]
```diff
@@ -268,7 +270,7 @@ func TestRequiredAffinitySingleNode(t *testing.T) {
 					PodAntiAffinityIn("service", "zone", []string{"securityscan", "value2"}, st.PodAntiAffinityWithRequiredReq).Obj(),
 			},
 			node: &node1,
-			wantStatus: framework.NewStatus(
```
Comment (sanposhiho) [inline pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:271]: This test case confirms even if no Affinity on incoming Pod, PreFilter won't return Skip if any existing Pod has AntiAffinity which match with incoming Pod.

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go [hunk 2]
```diff
@@ -540,16 +543,20 @@ func TestRequiredAffinitySingleNode(t *testing.T) {
 			p := plugintesting.SetupPluginWithInformers(ctx, t, New, &config.InterPodAffinityArgs{}, snapshot, namespaces)
 			state := framework.NewCycleState()
 			_, preFilterStatus := p.(framework.PreFilterPlugin).PreFilter(ctx, state, test.pod)
+			if preFilterStatus.Code() != test.wantPreFilterStatus.Code() {
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:546]: please use cmp.Diff instead, see https://github.com/kubernetes/kubernetes/blob/master/pkg/scheduler/framework/plugins/volumezone/volume_zone_test.go#L233-L240

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go [hunk 3]
```diff
@@ -1077,26 +1089,6 @@ func TestPreFilterStateAddRemovePod(t *testing.T) {
 		expectedAntiAffinity topologyToMatchedTermCount
 		expectedAffinity     topologyToMatchedTermCount
 	}{
-		{
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:1080]: why was this removed?

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go [hunk 4]
```diff
@@ -949,16 +958,19 @@ func TestRequiredAffinityMultipleNodes(t *testing.T) {
 				[]runtime.Object{
 					&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "NS1"}},
 				})
+			state := framework.NewCycleState()
+			_, preFilterStatus := p.(framework.PreFilterPlugin).PreFilter(ctx, state, test.pod)
+			if preFilterStatus.Code() != test.wantPreFilterStatus.Code() {
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:963]: ditto, use cmp.Diff as noted in the other comment, for both the PreFilter and Filter statues.

File: pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go (hunk 3 above)
Comment (sanposhiho) [reply pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go:1080]: It's because when no Affinity in incoming Pod, `AddPod` and `RemovePod` shouldn't be called.
They are skipped like Filter plugin. ([ref1](https://github.com/sanposhiho/kubernetes/blob/eabb70833a5649e10ab81f04423ce5cb16aba1b7/pkg/scheduler/framework/runtime/framework.go#L664), [ref2](https://github.com/sanposhiho/kubernetes/blob/eabb70833a5649e10ab81f04423ce5cb16aba1b7/pkg/scheduler/framework/runtime/framework.go#L699))

**Comment by sanposhiho** [comment]
@ahg-g Thanks for reviewing. Fixed.

**Comment by ahg-g** [comment]
Thanks, please squash.

**Comment by sanposhiho** [comment]
Squashed
